src/components/alert/alert.component.js

```diff
@@ -1,24 +1,43 @@
 import React from "react";
+import PropTypes from "prop-types";
+
 import Dialog from "../dialog";
-import AlertStyle from "./alert.style.js";
 
-class Alert extends Dialog {
-  componentTags(props) {
-    return {
-      "data-component": "alert",
-      "data-element": props["data-element"],
-      "data-role": props["data-role"],
-    };
-  }
+const Alert = ({ children, ...rest }) => (
+  <Dialog data-component="alert" {...rest}>
+    {children}
+  </Dialog>
+);
 
-  get closeIcon() {
-    return <AlertStyle>{super.closeIcon}</AlertStyle>;
-  }
-}
+Alert.propTypes = {
+  /** The ARIA role to be applied to the Dialog */
+  ariaRole: PropTypes.string,
+  /** Alert content */
+  children: PropTypes.node,
+  /** Controls the open state of the component */
+  open: PropTypes.bool,
+  /** A custom close event handler */
+  onCancel: PropTypes.func,
+  /** Determines if the Esc Key closes the Alert */
+  disableEscKey: PropTypes.bool,
+  /** Allows developers to specify a specific height for the dialog. */
+  height: PropTypes.string,
+  /** Title displayed at top of Alert */
+  title: PropTypes.oneOfType([PropTypes.string, PropTypes.object]),
+  /** Subtitle displayed at top of Alert */
+  subtitle: PropTypes.string,
+  /** Size of Alert */
+  size: PropTypes.string,
+  /** Determines if the close icon is shown */
+  showCloseIcon: PropTypes.bool,
```

Comment:
As we're already making a `BREAKING CHANGE`, I wonder whether we should remove `showCloseIcon` as the same functionality seems to be covered via whether a callback is passed to `onCancel`